### PR TITLE
Generate new private key if identity file doesn't exist

### PR DESCRIPTION
**Pull request by borzunov**

### CODE REVIEW

File: examples/albert/arguments.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ class BaseTrainingArguments:
         default=None,
         metadata={
             "help": "Path to a pre-generated private key file. If defined, makes the peer ID deterministic. "
-            "May be generated using ``./p2p-keygen`` from ``go-libp2p-daemon``."
+            "If the file does not exist, generates a new private key."
```
Comment (justheuristic) [inline examples/albert/arguments.py:41]: [nit] ... generates and saves?

File: hivemind/p2p/p2p_daemon.py [hunk 2]
```diff
@@ -113,8 +115,8 @@ async def create(
                          Details: https://pkg.go.dev/github.com/libp2p/go-libp2p-kad-dht#ModeOpt
         :param force_reachability: Force reachability mode (public/private)
         :param host_maddrs: Multiaddrs to listen for external connections from other p2p instances
-        :param identity_path: Path to a pre-generated private key file. If defined, makes the peer ID deterministic.
-                              May be generated using ``./p2p-keygen`` from ``go-libp2p-daemon``.
+        :param identity_path: Path to a private key file. If defined, makes the peer ID deterministic.
+                              If the file does not exist, generates a new private key.
```
Comment (justheuristic) [inline hivemind/p2p/p2p_daemon.py:119]: [nit] ... and saves it to that file

File: hivemind/p2p/p2p_daemon.py [hunk 3]
```diff
@@ -211,6 +217,15 @@ async def create(
...
         return self
 
+    @staticmethod
+    def generate_identity(identity_path: str) -> None:
+        private_key = RSAPrivateKey()
+        protobuf = crypto_pb2.PrivateKey(key_type=crypto_pb2.KeyType.RSA, data=private_key.to_bytes())
+
+        with open(identity_path, "wb") as f:
+            f.write(protobuf.SerializeToString())
+        os.chmod(identity_path, 0o400)
```
Comment (justheuristic) [inline hivemind/p2p/p2p_daemon.py:227]: awesome!

**Review by justheuristic — APPROVED** [review]

File: hivemind/p2p/p2p_daemon.py [hunk 4]
```diff
@@ -165,7 +167,11 @@ async def create(
         ]:
             if value:
                 process_kwargs[param] = self._maddrs_to_str(value)
+
         if identity_path is not None:
+            if not os.path.isfile(identity_path):
```
Comment (justheuristic) [inline hivemind/p2p/p2p_daemon.py:172]: [nit] there may be a situation where not only file is missing, but the preceding directory is also nonexistent. Here's what can happen:
- create folders (os.mkdirs(..., exist_ok=True))
- raise an explicit error message ("e.g. parent directory does not exist, not "file not found"")

Comment (borzunov) [reply hivemind/p2p/p2p_daemon.py:172]: I've added the explicit error message to `P2P.generate_identity_file()`.

**Pull request merged**